jwst/badpix_selfcal/badpix_selfcal.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+import numpy as np
+import jwst.datamodels as dm
+from jwst.outlier_detection.outlier_detection_ifu import medfilt
+from stdatamodels.jwst.datamodels.dqflags import pixel
+
+
+def badpix_selfcal(minimg: np.ndarray,
+                   flagfrac_lower: float = 0.001,
+                   flagfrac_upper: float = 0.001,
+                   kernel_size: int = 15,
+                   dispaxis=None) -> np.ndarray:
+    """
+    Flag residual artifacts as bad pixels in the DQ array of a JWST exposure
+
+    Parameters
+    ----------
+    minimg: np.ndarray
+        Selfcal data of shape (x, y), i.e., after some operation has
+        already been taken to combine multiple exposures,
+        typically a MIN operation.
+    flagfrac_lower: float
+        Fraction of pixels to flag on the low end
+    flagfrac_upper: float
+        Fraction of pixels to flag on the high end
+    kernel_size: int
+        Size of kernel for median filter
+    dispaxis: int
+        Dispersion axis, either 1 or 2. If None, a two-dimensional
+        median filter is applied.
+
+    Returns
+    -------
+    flagged_indices: np.ndarray
+        Indices of the flagged pixels,
+        shaped like output from np.where
+    """
+    if (flagfrac_lower < 0) or (flagfrac_upper < 0):
+        raise ValueError("Flagged fraction must be between 0 and 0.5")
+    if flagfrac_lower + flagfrac_upper >= 1:
+        raise ValueError("Total flagged fraction (lower plus upper) must be less than 1")
+
+    if dispaxis not in [1, 2, None]:
+        raise ValueError("dispaxis must be either 1 or 2, or None.")
+
+    # Determine outliers using median filter
+    if dispaxis is None:
+        kern_size = (kernel_size, kernel_size)
+    elif dispaxis == 2:
+        # check that this is the right way around!
+        kern_size = (kernel_size, 1)
+    elif dispaxis == 1:
+        kern_size = (1, kernel_size)
+
+    smoothed = medfilt(minimg, kern_size)
```

Comment:
This line sometimes raises `RuntimeWarning: All-NaN slice encountered` for NaNs in the data, but that's a normal condition for NIRSpec IFU data at least.  Can we please catch and filter the warning?